core/domain/exp_services.py

```diff
@@ -792,6 +793,18 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (exploration_model.version, exploration.version))
 
+    old_states_dicts = copy.deepcopy(exploration_model.states)
```

Comment:
Do all these stuff only if ml classifiers are enabled globally. I think we already have a constant for it in feconf.